packages/mimir/src/rules/parameter-properties.ts

```diff
@@ -0,0 +1,193 @@
+    }
+
+    private get failureString(): string {
+        switch (this.options.mode) {
+            case 'never':
+                return 'Parameter properties have been disallowed.';
+            case 'when-possible':
+                return 'Use parameter properties when possible.';
+            default:
+                return 'Either all constructor parameters must be parameter properties, or none.';
+        }
+    }
+
+    private checkConstructorParameters(
+        construct: ts.ConstructorDeclaration,
+        classNode: ts.ClassDeclaration | ts.ClassExpression,
+    ): void {
+        switch (this.options.mode) {
+            case 'never':
+                for (const param of construct.parameters.filter(isParameterProperty))
+                    this.addFailureAtNode(
+                        param,
+                        this.failureString,
+                        getFixerForDisallowedParameterProp(
+                            construct,
+                            param,
+                            getLineBreakStyle(this.context.sourceFile),
+                        ),
+                    );
+                break;
+            case 'when-possible':
+                for (const param of construct.parameters.filter(
+                    (p: ts.ParameterDeclaration) => !isParameterProperty(p) && canBeParameterProperty(p, construct),
+                ))
+                    this.addFailureAtNode(
+                        param,
+                        this.failureString,
+                        getFixerForLonghandProp(param, construct, classNode),
+                    );
+                break;
+            default:
+                if (construct.parameters.some(isParameterProperty) && !construct.parameters.every(isParameterProperty))
+                    for (const param of construct.parameters.filter(isParameterProperty))
+                        this.addFailure(
+                            construct.parameters.pos,
+                            construct.parameters.end,
+                            this.failureString,
+                            getFixerForDisallowedParameterProp(
+                                construct,
+                                param,
+                                getLineBreakStyle(this.context.sourceFile),
+                            ),
+                        );
+        }
+    }
+}
+
+function canBeParameterProperty(param: ts.ParameterDeclaration, construct: ts.ConstructorDeclaration): boolean {
+    const stmts = Array.from(construct.body!.statements);
+    if (getSuperCall(construct)) stmts.shift();
+    for (const stmt of stmts) {
+        /**
+         * See https://github.com/fimbullinter/wotan/issues/167#issuecomment-378945862 for details on this condition
+         */
+        if (
+            !isExpressionStatement(stmt) ||
+            !isBinaryExpression(stmt.expression) ||
+            !(stmt.expression.operatorToken.kind === ts.SyntaxKind.EqualsToken) ||
```

Comment:
could you remove the parens on this line. they made me think it belongs to the condition on the next line